server/webhook.go

```diff
@@ -34,7 +34,23 @@ func getStringFromTemplate(tmpl string, data any) (string, error) {
 	return result.String(), nil
 }
 
+func (p *Plugin) authenticateWebhook(r *http.Request) error {
+	token := r.Header.Get("X-MM-Cloud-Plugin-Auth-Token")
+
+	if token != p.configuration.ProvisioningServerWebhookSecret {
```

Comment:
0/5 might want to use `crypto/subtle.ConstantTimeCompare` here, but it's not a big deal.